_posts/2016-03-24-app-analytics-redux.markdown

```diff
@@ -0,0 +1,93 @@
+
+The middleware installation instructions currently assume you're building a fat client JS app and will be using [`Analytics.js`](https://github.com/segmentio/analytics.js). But you could just as easily swap in [`analytics-node`](https://github.com/segmentio/analytics-node) if you're building a Node app instead. Using `redux-thunk` for async data flow? No problem, the middleware supports that too.
+
+## Connecting with Segment
+
+If you've already signed up with an account and **create a _Project_ for your app** in Segment all you need to do is `_.plop` the `WRITE_KEY` provided into your app while initializing, and let the Redux middleware do the rest. The key is safe to share, so you don't need to worry about [locking it up](http://technologyadvice.github.io/lock-up-your-customer-accounts-give-away-the-key/).
+
+## Sending an event
+
+Once you've installed the `redux-segment` middleware and connected with Segment you can start tracking events immediately. As mentioned earlier, users of React Router Redux will enjoy out of the box support and will see _Page_ events start flying the moment they nagivate between routes. Additional events can be set-up with just a few lines of code in your reducer.
+
+Here's an example _Identify_ event we use to track user login events on some of our React apps here at TA:
+
+```js
+import { EventTypes } from 'redux-segment'
+
+export const loginSuccess = (token, user) => ({
+  type: SESSION_LOGIN_SUCCESS,
+  payload: {
+    token,
+    user,
+  },
+  meta: {
+    analytics: {
+      eventType: EventTypes.identify,
+      eventPayload: {
+        userId: user.id,
+        traits: (() => {
+          const { email, firstName, imageUrl, lastName, phoneNumber } = user
+          return {
+            avatar: imageUrl,
+            email,
+            firstName,
+            lastName,
+            phone: phoneNumber,
+          }
+        })(),
+      },
+    },
+  },
+})
+```
+
+Notice how we've taken a simple reducer and added a `meta` property called `analytics`, along with an event type and payload. That's all there is to it! See the middleware [Usage section](https://github.com/rangle/redux-segment#usage) for different event types, supported routers and additional documentation.
+
+## Track to your heart's content
+
+Now that you have an arsenal of analytics tooling integrations at your fingertips you can ask your stakeholders what they want to measure, instead of asking them how they want to go about measuring it.
+
+I've used Segment in the past to set-up and scale several single-page apps tracked by Google Analytics, track application errors with Sentry and watch users navigate a site to discover UX issues. Currently I'm working on an AWS client-side monitoring solution leveraging [Lambda](https://aws.amazon.com/lambda/) and webhooks. And I get geeked every time someone shows me some new, so please don't hesitate to gush about your favorite tools in the comments section below.
+
+[Keen.io]: https://keen.io/
```

Comment:
@TomFrost may have noticed how his writing style influenced the way I opened up the article.